Assets/BossRoom/Scripts/Server/Game/Entity/ServerWaveSpawner.cs

```diff
@@ -217,20 +229,45 @@ NetworkObject SpawnPrefab()
             {
                 clone.Spawn();
             }
+
+            if (m_SpawnedEntityDetectDistance > -1)
+            {
+                // need to override the spawned creature's detection range (if they even have a detection range!)
+                var serverChar = clone.GetComponent<ServerCharacter>();
+                if (serverChar)
+                {
+                    serverChar.SetDetectRadius(m_SpawnedEntityDetectDistance);
+                }
+            }
+
             return clone;
         }
 
         bool IsRoomAvailableForAnotherSpawn()
         {
-            if (m_MaxActiveSpawns <= 0)
-            {
-                // no max-spawn limit
-                return true;
-            }
             // references to spawned components that no longer exist will become null,
             // so clear those out. Then we know how many we have left
             m_ActiveSpawns.RemoveAll(spawnedNetworkObject => { return spawnedNetworkObject == null; });
-            return m_ActiveSpawns.Count < m_MaxActiveSpawns;
+            return m_ActiveSpawns.Count < GetCurrentSpawnCap();
+        }
+
+        /// <summary>
+        /// Returns the current max number of entities we should try to maintain.
+        /// This can change based on the current number of living players; if the cap goes below
+        /// our current number of active spawns, we don't spawn anything new until we're below the cap.
+        /// </summary>
+        int GetCurrentSpawnCap()
+        {
+            int numPlayers = 0;
```

Comment:
Just to confirm, we are not despawning the surplus of Active Spawns, when the player leaves the server/host session?

Reply:
Correct, my understanding is the monsters shouldn't de-spawn, we just don't create more while we're over the cap.

Reply:
Thanks for confirming!